pilot/pkg/serviceregistry/serviceentry/controller.go

```diff
@@ -512,41 +521,55 @@ func (s *Controller) WorkloadInstanceHandler(wi *model.WorkloadInstance, event m
 	fullPush := false
 	for _, cfg := range cfgs {
 		se := cfg.Spec.(*networking.ServiceEntry)
-		if se.WorkloadSelector == nil || !labels.Instance(se.WorkloadSelector.Labels).SubsetOf(wi.Endpoint.Labels) {
-			// Not a match, skip this one
+
+		if se.WorkloadSelector == nil || (!labelsChanged && !labels.Instance(se.WorkloadSelector.Labels).SubsetOf(wi.Endpoint.Labels)) {
```

Comment:
how can this happend 
!labelsChanged && !labels.Instance(se.WorkloadSelector.Labels).SubsetOf(wi.Endpoint.Labels)